### PR TITLE
Fix update-openapi-spec to not change caller CWD

### PR DESCRIPTION
**Pull request by thockin**
I suspect tests pass only because they don't keep enough state in-between runs.

Short story: shell functions that use `cd` or `pushd`/`popd` is suspect.  Make sure you don't change the caller's working dir.

/kind bug

```release-note
NA
```

### CODE REVIEW

File: hack/lib/etcd.sh [hunk 1]
```diff
@@ -144,42 +144,46 @@ kube::etcd::cleanup() {
...
-    download_file="etcd-v${ETCD_VERSION}-${os}-${arch}.tar.gz"
-    kube::util::download_file "${url}" "${download_file}"
-    tar xzf "${download_file}"
-    ln -fns "etcd-v${ETCD_VERSION}-${os}-${arch}" etcd
-    rm "${download_file}"
-  else
-    kube::log::info "${os} is NOT supported."
-  fi
-  kube::log::info "etcd v${ETCD_VERSION} installed. To use:"
-  kube::log::info "export PATH=\"$(pwd)/etcd:\${PATH}\""
```
Comment (aojea) [inline hack/lib/etcd.sh:181]: I find these logs useful, you removed  all of them

Comment (thockin) [reply hack/lib/etcd.sh:181]: a) They clutter the "normal path" screen with information I didn't ask for and usually don't need
b) They make it look like I need to do something (e.g. "cut and paste the `export`")
c) The third one actually tells me to do exactly that

How about I V=3 them?

**Comment by pacoxu** [comment]
/sig etcd 
(not sure)
/retest

**Comment by BenTheElder** [comment]
Thanks.

File: hack/lib/etcd.sh (hunk 1 above)
Comment (gjkim42) [reply hack/lib/etcd.sh:181]: Actually, cut and paste is needed if you want to execute hack/local-up-cluster.sh after this.

If you want to build a local cluster,
1. ./hack/install-etcd.sh
2. export PATH=to this PATH
3. ./hack/local-up-cluster.sh

**Pull request merged**